blog/md/ref/jenkins/introToJenkins.md

```diff
@@ -0,0 +1,176 @@
+
+docker run -p 8080:8080 -p 50000:50000 jenkins/jenkins:lts
+
+After pulling the image and starting the Jenkins server, I accessed Jenkins through the browser using the localhost URL.
+
+![Getting started: Instance configuration](https://miro.medium.com/v2/resize:fit:1100/format:webp/0*fEjtkre6udi1mYIa)
+
+After finishing the initial setup, I created my first Jenkins job, which was a freestyle project that simply echoed a message.
+
+“Hello from Jenkins!”
+
+![Console output](https://miro.medium.com/v2/resize:fit:1100/format:webp/0*3DTBdONl-XruyZ_V)
+
+The console output confirmed success. This was my first working Jenkins job!
+
+## Day 2 – Maven Pipeline Configuration & GitHub Integration
+
+On Day 2, I used a Maven-based Java project from GitHub to go from freestyle jobs to scripted pipelines.
+
+In Jenkins, I made a `Build-Maven-App` Pipeline project and connected it to my GitHub repository.
+
+GitHub Repo: <https://github.com/iiamolu/simple-java-maven-app>
+
+![Console output](https://miro.medium.com/v2/resize:fit:1100/format:webp/0*gYw7oxetIpz8N9rf)
+
+Created a Jenkinsfile and added it to my repo. It contained three main stages: Build, Test, and Deliver.
+
+`pipeline {`
+
+`agent any`
+
+`options {`
+
+`skipStagesAfterUnstable()`
+
+`}`
+
+`tools {`
+
+`maven ‘Maven 3.9.6’`
+
+`}`
+
+`stages {`
+
+`stage(‘Build’) {`
+
+`steps {`
+
+`sh ‘mvn clean compile’`
+
+`}`
+
+`}`
+
+`stage(‘Test’) {`
+
+`steps {`
+
+`sh ‘mvn test’`
+
+`}`
+
+`post {`
+
+`always {`
+
+`junit ‘target/surefire-reports/*.xml’`
+
+`}`
+
+`}`
+
+`}`
+
+`stage(‘Package’) {`
+
+`steps {`
+
+`sh ‘mvn package’`
+
+`}`
+
+`}`
+
+`stage(‘Deliver’) {`
+
+`steps {`
+
+`sh ‘./jenkins/scripts/deliver.sh’`
+
+`}`
+
+`}`
+
+`}`
+
+`}`
+
+![My three failed attempts...](https://miro.medium.com/v2/resize:fit:786/format:webp/1*h2sBSI_hZms6VdAukapTfg.png)
+
+After some troubleshooting (including fixing naming and Maven version issues), I successfully built the project and produced a .jar file.
+
+![Build #4 status](https://miro.medium.com/v2/resize:fit:1100/format:webp/1*enVxCjmZfr0agSH0YqS5-g.png)
+
+## Day 3 – Testing, Debugging & Final Pipeline
+
+Concentrated on incorporating automated testing into the pipeline on day three. There was already a JUnit-written test class in my Maven project. I modified the Jenkinsfile to execute `mvn test` and use the JUnit plugin to report the outcomes.
+
+![Jenkinsfile](https://miro.medium.com/v2/resize:fit:1100/format:webp/1*WA-EZCYSJfyeN0YDNFNcpg.png)
+
+![Jenkinsfile2](https://miro.medium.com/v2/resize:fit:1100/format:webp/1*q7PnUlyYrLEPGRtGvXR91g.png)
+
+Here is the test stage I added to my Jenkinsfile:
+
+stage(‘Test’) {
+steps {
+sh ‘mvn test’
+}
+post {
+always {
+junit ‘target/surefire-reports/TEST-*.xml’
+}
+}
+}
+
+After building the pipeline, I verified that the test results were collected and displayed under my Jenkins job.
+
+![Test results](https://miro.medium.com/v2/resize:fit:1100/format:webp/0*qUZ4fNd2kH_puksI)
+
```

Comment:
we would need to paste the png into the document and make them in the project_root/images/dev_ops and link the relative path into this doc.